hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -241,6 +241,18 @@ public SessionImpl(SessionFactoryImpl factory, SessionCreationOptions options) {
 			setHibernateFlushMode( initialMode );
 		}
 
+		if ( factory.getDefinedFilterNames().contains("_tenantId") ) {
```

Comment:
IMO `"_tenantId"` should be moved into a constant, just like the parameter name `tenantId`